src/App.tsx

```diff
@@ -15,47 +15,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      return !isReversed
+        ? visibleGoods.sort()
+        : visibleGoods.sort((a, b) => (a > b ? -1 : 1));
+    case SortType.LENGTH:
+      return !isReversed
+        ? visibleGoods.sort((a, b) => a.length - b.length)
+        : visibleGoods.sort((a, b) => a.length - b.length).reverse();
+    case SortType.NONE:
+      return isReversed ? visibleGoods.reverse() : visibleGoods;
```

Comment:
`.reverse()` function repeated several times 

you can move check `isReverse` under the switch and reverse list if it necessary after sorting in the switch, in this way - 
```

if(isReversed){
  visibleGoods.reverse();
}
```